src/common/datatypes/Vector.h

```diff
@@ -0,0 +1,171 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#ifndef COMMON_DATATYPES_VECTOR_H_
+#define COMMON_DATATYPES_VECTOR_H_
+#include <folly/dynamic.h>
+
+#include <cassert>
+#include <cmath>
+#include <iostream>
+#include <sstream>
+#include <vector>
+namespace nebula {
+/*
+ * Vector Type is consist of a vector of values and dimension of vector.
+ * The element type in vector is float.
+ */
+struct Vector {
+  int64_t dimension;
+  std::vector<float> values;
+
+  Vector() : dimension(0) {}
+  Vector(const Vector& rhs) = default;
+  Vector(Vector&& rhs) noexcept = default;
+  explicit Vector(std::vector<float> vals, int64_t dim) : dimension(dim), values(std::move(vals)) {}
+  explicit Vector(std::vector<float> vals) : dimension(vals.size()), values(std::move(vals)) {}
+
+  Vector& operator=(const Vector& rhs) {
+    if (this == &rhs) {
+      return *this;
+    }
+    values = rhs.values;
+    dimension = rhs.dimension;
+    return *this;
+  }
+
+  Vector& operator=(Vector&& rhs) noexcept {
+    if (this == &rhs) {
+      return *this;
+    }
+    values = std::move(rhs.values);
+    dimension = rhs.dimension;
+    return *this;
+  }
+
+  size_t dim() const {
+    return dimension;
+  }
+
+  std::vector<float> data() const {
+    return values;
+  }
+
+  inline static bool floatcmp(float a,
+                              float b,
+                              double relative_tolerance = 1e-6,
+                              double absolute_tolerance = 1e-6) {
+    if (a == b) {
+      return true;
+    }
+    double diff = std::fabs(a - b);
+    if (diff <= absolute_tolerance) {
+      return true;
+    }
+    return diff <= relative_tolerance * std::max(std::fabs(a), std::fabs(b));
+  }
+
+  bool operator==(const Vector& rhs) const {
+    if (dimension != rhs.dimension) {
+      return false;
+    }
+    for (size_t i = 0; i < static_cast<size_t>(dimension); ++i) {
+      if (!floatcmp(values[i], rhs.values[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool operator<(const Vector& rhs) const {
+    if (dimension != rhs.dimension) {
+      return dimension < rhs.dimension;
+    }
+    return values < rhs.values;
+  }
+
+  bool operator>(const Vector& rhs) const {
+    if (dimension != rhs.dimension) {
+      return dimension > rhs.dimension;
+    }
+    return values > rhs.values;
+  }
+
+  Vector operator+(const Vector& rhs) const {
+    Vector result;
+    result.dimension = dimension;
+    result.values.resize(dimension);
+    for (size_t i = 0; i < static_cast<size_t>(dimension); ++i) {
+      result.values[i] = values[i] + rhs.values[i];
+    }
+    return result;
+  }
+
+  Vector operator-(const Vector& rhs) const {
+    Vector result;
+    result.dimension = dimension;
+    result.values.resize(dimension);
+    for (size_t i = 0; i < static_cast<size_t>(dimension); ++i) {
+      result.values[i] = values[i] - rhs.values[i];
+    }
```

Comment:
Comparison operations on vector data should not be allowed. We should limit its operation semantics, which is also the key reason why vector data type cannot reuse list type.